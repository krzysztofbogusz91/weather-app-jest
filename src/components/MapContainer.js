import React, {Component} from 'react';
import {Map, Marker, GoogleApiWrapper} from 'google-maps-react';

//google map react documentation: https://github.com/fullstackreact/google-maps-react

export class MapContainer extends Component {
    constructor(props) {
        super(props);
        this.state={
            selectedPlace: {name: "Map"},
            lat: this.props.lat,
            lng: this.props.lng
        }
    }

    render() {
        //first div styles map
        return (
           <div style={{position:"relative", width:"100%", height:"200px"}}>
            <Map
                style={{width: '100%', height: '100%', position: 'relative'}}
                google={this.props.google}
                center={{
                    lat: this.props.lat ,
                    lng: this.props.lng
                }}
                zoom={13}>

                <Marker
                    position={{lat: this.props.lat, lng:this.props.lng}}
                    name={'Current location'}
                            />
            </Map>
           </div>
        );
    }
}

export default GoogleApiWrapper({
    apiKey: "AIzaSyBSjHCxvrOJ9vPn9ReAQOchfXblNATQtf8"
})(MapContainer)