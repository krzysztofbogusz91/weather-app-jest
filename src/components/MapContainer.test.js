import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import set from '../enzyme_set_up';
import { MapContainer } from './MapContainer';

describe('MapContainer',()=>{
    const props = {lat: 51, lng: 16}
    const us = shallow( <MapContainer {...props} /> )

    it('renders without crashing', () => {
        expect(us).toMatchSnapshot();
    });
      
})
