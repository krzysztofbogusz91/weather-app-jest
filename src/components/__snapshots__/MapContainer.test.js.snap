// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapContainer renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapContainer
    lat={51}
    lng={16}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Map
        center={
                Object {
                        "lat": 51,
                        "lng": 16,
                      }
        }
        centerAroundCurrentLocation={false}
        containerStyle={Object {}}
        google={undefined}
        initialCenter={
                Object {
                        "lat": 37.774929,
                        "lng": -122.419416,
                      }
        }
        style={
                Object {
                        "height": "100%",
                        "position": "relative",
                        "width": "100%",
                      }
        }
        visible={true}
        zoom={13}
>
        <Marker
                name="Current location"
                position={
                        Object {
                                "lat": 51,
                                "lng": 16,
                              }
                }
        />
</Map>,
      "style": Object {
        "height": "200px",
        "position": "relative",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "center": Object {
          "lat": 51,
          "lng": 16,
        },
        "centerAroundCurrentLocation": false,
        "children": <Marker
          name="Current location"
          position={
                    Object {
                              "lat": 51,
                              "lng": 16,
                            }
          }
/>,
        "containerStyle": Object {},
        "google": undefined,
        "initialCenter": Object {
          "lat": 37.774929,
          "lng": -122.419416,
        },
        "style": Object {
          "height": "100%",
          "position": "relative",
          "width": "100%",
        },
        "visible": true,
        "zoom": 13,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "name": "Current location",
          "position": Object {
            "lat": 51,
            "lng": 16,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Map
          center={
                    Object {
                              "lat": 51,
                              "lng": 16,
                            }
          }
          centerAroundCurrentLocation={false}
          containerStyle={Object {}}
          google={undefined}
          initialCenter={
                    Object {
                              "lat": 37.774929,
                              "lng": -122.419416,
                            }
          }
          style={
                    Object {
                              "height": "100%",
                              "position": "relative",
                              "width": "100%",
                            }
          }
          visible={true}
          zoom={13}
>
          <Marker
                    name="Current location"
                    position={
                              Object {
                                        "lat": 51,
                                        "lng": 16,
                                      }
                    }
          />
</Map>,
        "style": Object {
          "height": "200px",
          "position": "relative",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "center": Object {
            "lat": 51,
            "lng": 16,
          },
          "centerAroundCurrentLocation": false,
          "children": <Marker
            name="Current location"
            position={
                        Object {
                                    "lat": 51,
                                    "lng": 16,
                                  }
            }
/>,
          "containerStyle": Object {},
          "google": undefined,
          "initialCenter": Object {
            "lat": 37.774929,
            "lng": -122.419416,
          },
          "style": Object {
            "height": "100%",
            "position": "relative",
            "width": "100%",
          },
          "visible": true,
          "zoom": 13,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "name": "Current location",
            "position": Object {
              "lat": 51,
              "lng": 16,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
